Framework/API/test/MatrixWorkspaceTest.h

```diff
@@ -1402,8 +1433,26 @@ class MatrixWorkspaceTestPerformance : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(result, 5214709.740869, 1e-6);
   }
 
+  void test_calculateL2() {
+
+    /*
+     * Simulate the L2 calculation performed via the Workspace/Instrument
+     * interface.
+     */
+    auto instrument = m_sansTestWorkspace.getInstrument();
+    auto sample = instrument->getSample();
+    double l2 = 0;
+    for (size_t i = 0; i < m_sansTestWorkspace.getNumberHistograms(); ++i) {
+      auto detector = m_sansTestWorkspace.getDetector(i);
+      l2 += detector->getDistance(*sample);
```

Comment:
@OwenArnold If I remember correctly there is a caching mechanism for the positions of parent components. Do you know if the cache would be used in the case of this instrument? If so, the benchmark here is basically also measuring the time to initialize the cache. Have you tried adding a second test that does exactly the same thing again, to see if the performance is significantly different?